tests/python-tests.ps1

```diff
@@ -88,6 +94,30 @@ Describe "Tests" {
         It "Check if shared libraries are linked correctly" {
             "bash ./sources/psutil-install-test.sh" | Should -ReturnZeroExitCode
         }
+
+        It "Relocatable Python" {
+            $semversion = [semver] $Version
+            $pyfilename = "python$($semversion.Major).$($semversion.Minor)"
+            $artifactPath = Join-Path "Python" $Version | Join-Path -ChildPath $Architecture
+
+            $relocatedPython = Join-Path $HOME "relocated_python"
+            $relocatedPythonTool = Join-Path -Path $relocatedPython -ChildPath $artifactPath
+            $relocatedFullPath = Join-Path $relocatedPythonTool "bin" | Join-Path -ChildPath $pyfilename
+
+            # copy the current build to relocated_python
+            $toolCacheArtifact = Join-Path $env:RUNNER_TOOL_CACHE $artifactPath
+            moveAssets -source $toolCacheArtifact -destination $relocatedPythonTool
+            try {
+                # Verify that relocated Python works
+                $relocatedFullPath | Should -Exist
+                "$relocatedFullPath --version" | Should -ReturnZeroExitCode
+                "sudo $relocatedFullPath --version" | Should -ReturnZeroExitCode
+            }
+            finally {
+                # Revert the changes for other tests
+                moveAssets -source $relocatedPythonTool -destination $toolCacheArtifact
+            }
+        }
```

Comment:
I think this might fail on macOS.
@jaswanthikolla, can you run the full build in your fork to make sure this passes ?
It might save some time if it requires some modifications before maintainers validate the workflow / review the PR.

Reply:
This is Ubuntu/Linux Specific code, So this test doesn't get executed for macOS. May be, I can simplify by removing powershell code and write bash instead.

Reply:
It does fail on macOS which is an UNIX (it's UNIX specific code, not Linux): https://github.com/mayeut/python-versions/actions/runs/11766830107/job/32774981352